### PR TITLE
[Federation] Call the federation cluster log dump script on failure

### PR DESCRIPTION
**Pull request by perotinus**
Companion to #43028.

### CODE REVIEW

**Comment by madhusudancs** [comment]
@krzyzacy oh thanks! Did not know this existed!

File: kubetest/e2e.go [hunk 1]
```diff
@@ -161,6 +163,9 @@ func run(deploy deployer, o options) error {
 		errs = appendError(errs, xmlWrap("DumpClusterLogs", func() error {
 			return DumpClusterLogs(o.dump)
 		}))
+		errs = appendError(errs, xmlWrap("DumpFederationClusterLogs", func() error {
```
Comment (madhusudancs) [inline kubetest/e2e.go]: s/DumpFederationClusterLogs/DumpFederationLogs/

File: kubetest/e2e.go [hunk 2]
```diff
@@ -105,7 +105,9 @@ func run(deploy deployer, o options) error {
 		}
 		if o.federation {
 			if err := xmlWrap("Federation Up", FedUp); err != nil {
-				// TODO: Dump federation related logs.
+				xmlWrap("DumpFederationClusterLogs", func() error {
```
Comment (madhusudancs) [inline kubetest/e2e.go]: s/DumpFederationClusterLogs/DumpFederationLogs/

File: kubetest/e2e.go [hunk 3]
```diff
@@ -105,7 +105,9 @@ func run(deploy deployer, o options) error {
...
+					return DumpFederationClusterLogs(o.dump)
```
Comment (madhusudancs) [inline kubetest/e2e.go]: Same here.

File: kubetest/e2e.go [hunk 4]
```diff
@@ -381,6 +386,11 @@ func DumpClusterLogs(location string) error {
 	return finishRunning(exec.Command("./cluster/log-dump.sh", location))
 }
 
+func DumpFederationClusterLogs(location string) error {
+	log.Printf("Dumping Federation cluster logs to: %v", location)
```
Comment (madhusudancs) [inline kubetest/e2e.go]: "Dumping Federation logs ..."

File: kubetest/e2e.go (hunk 2 above)
Comment (perotinus) [inline kubetest/e2e.go]: Done.

File: kubetest/e2e.go (hunk 3 above)
Comment (perotinus) [inline kubetest/e2e.go]: Done.

File: kubetest/e2e.go (hunk 1 above)
Comment (perotinus) [inline kubetest/e2e.go]: Done.

File: kubetest/e2e.go (hunk 4 above)
Comment (perotinus) [inline kubetest/e2e.go]: Done.

**Pull request merged**